### ISSUE TITLE
Upgrade Meteor Testapp to Meteor 3.0 (Beta)

### ISSUE THREAD

**Issue by jankapunkt** [sandstorm-io/sandstorm#3709]
Meteor is closing in to [a new major release](https://guide.meteor.com/3.0-migration) with lots of breaking changes. Is this something that is relevant to this project?

**Comment by ocdtrekkie** [comment]
We are currently stuck a bit further behind than that, but it's definitely a goal.

**Comment by xet7** [comment]
Kind of related, I mentioned Sandstorm at podcast today https://www.youtube.com/watch?v=zic-h8jG6F8

**Comment by jankapunkt** [comment]
I'll give it a try on updating it to the Beta and see what issues arise in the process.

**Comment by ocdtrekkie** [comment]
@jankapunkt So the main thing I can already tell you is the standing blocker is that we use MongoDB 2.6.

**Comment by jankapunkt** [comment]
The Migration from Node 14 will be to 20 with Meteor 3 as they decided to keep up with the Node.js lifecycle as good as possible in order to get all the security fixes and features etc.

However, the bigger part of the Migration is actually the removal of Fibers (coroutines) as they are not supported by Node >=16 and therefore all the internal Fibers-related code is migrated to async/await. Most of all this affects nearly all functionality where Mongo.Collection is involved or where Fibers has been in direct usage.

A detailed list of what's affected is collected here: https://github.com/meteor/meteor/blob/d5c3b2eeafd0ad78ee7e2553f3f269c5c2a2e2a9/docs/generators/changelog/versions/3.0.md#L5-L17

Note this list only covers Meteor core and core-packages (like Accounts etc.) but no community packages.

Regarding the MongoDB migration I can actually help out with an article I wrote, after I wrestled with this when migrating from 4 to 5: https://dev.to/jankapunkt/meteorjs-and-meteor-up-mongodb-migration-to-a-new-major-version-40b7

However, the article is Meteor+Meteor-Up focused.

**Comment by mnutt** [comment]
This will happen as part of this PR: https://github.com/sandstorm-io/sandstorm/pull/3780

**Comment by ocdtrekkie** [comment]
Yeah, I think we can close this issue since we have a resolving PR in the wings.
